bindings/go/src/examples/perftest/perftest.go

```diff
@@ -232,6 +230,10 @@ func serverAmRecvHandler(header unsafe.Pointer, headerSize uint64, data *UcpAmDa
 	if data.IsDataValid() {
 		atomic.AddUint32(&perfTest.numCompletedRequests, 1)
 	} else {
+		if perfTestParams.memType == UCS_MEMORY_TYPE_CUDA && !CudaCheckCurrentContext() {
+			CudaSetDevice();
```

Comment:
minor: WDYT about checking `CudaCheckCurrentContext` or `cuCtxGetDevice` during `CudaSetDevice` and do nothing if context (or device) is set?

Reply:
I would like to check for `CudaSetDevice` result and call `panic` in case of error (here and inside `serverStart`). Or even always do panic in case if `CudaSetDevice` fails.

Reply:
Done.